### PR TITLE
Add conditionals for cloud_provider in SSH configs

### PR DESCRIPTION
**Pull request by stencell**
##### SUMMARY
With multiple cloud providers now being regularly used, we want to have more nuanced SSH configs for each of them vs lumping them all into one. 

This introduces basic conditionals to preserve functionality that we have had for `aws` environments as well as newer `osp` environments.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- bastion
- bastion-lite
- set_env_authorized_key

##### ADDITIONAL INFORMATION
@fridim @miteshrh @prakhar1985 
If you guys want to add anything here for additional host entries with the `osp` provider, please do so start with the changes in this branch as the basics.

### CODE REVIEW

File: ansible/roles/bastion-lite/files/bastion_ssh_config.j2 [hunk 1]
```diff
@@ -9,3 +10,17 @@ Host ec2* *.internal *.example.com
   StrictHostKeyChecking no
   ConnectTimeout 60
   ConnectionAttempts 10
+{% endif %}	
+{% if cloud_provider == 'osp' %}	
+Host *.example.com
+  User {{remote_user}}	
+{% if use_own_key|bool %}
```
Comment (jkupferer) [inline ansible/roles/bastion-lite/files/bastion_ssh_config.j2]: Can we add some indentation in the template to help trace the nested conditions? Something like:

```
{% if cloud_provider == 'osp' %}
...
{%   if use_own_key|bool %}
...
{%   else %}
...
{%   endif %}
...
{% endif %}
```

File: ansible/roles/bastion-lite/files/bastion_ssh_config.j2 [hunk 2]
```diff
@@ -9,3 +10,17 @@ Host ec2* *.internal *.example.com
   StrictHostKeyChecking no
   ConnectTimeout 60
   ConnectionAttempts 10
+{% endif %}
```
Comment (jkupferer) [inline ansible/roles/bastion-lite/files/bastion_ssh_config.j2]: `{% elif cloud_provider == 'osp' %}` maybe?

**Comment by fridim** [comment]
It's weird that a template lives in a `files` directory. Should we move the `.j2` templates into the `templates/` directory now ?

**Comment by stencell** [comment]
I think this is good to merge now. @jkupferer after you review, please do the magic.

**Pull request merged**